synapse/notifier.py

```diff
@@ -444,15 +444,23 @@ def on_new_event(
 
             self.notify_replication()
 
-            # Notify appservices.
-            try:
-                self.appservice_handler.notify_interested_services_ephemeral(
-                    stream_key,
-                    new_token,
-                    users,
-                )
-            except Exception:
-                logger.exception("Error notifying application services of event")
+            # Notify appservices of updates in ephemeral event streams.
+            # Only the following streams are currently supported.
+            if stream_key in (
+                "typing_key",
+                "receipt_key",
+                "presence_key",
+                "to_device_key",
+            ):
```

Comment:
I didn't really understand the logic for moving this out of `appservice_handler`. Isn't the choice of what streams to care about a matter for the appservice handler, not the notifier?